pkg/pillar/cmd/zedrouter/server.go

```diff
@@ -227,3 +253,238 @@ func (hdl hostnameHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		w.Write(resp)
 	}
 }
+
+// ServeHTTP for metadataHandler process requests for ec2-compatible requests
+func (hdl metadataHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	log.Errorf("ServeHTTP metadataHandler %s", r.URL.Path)
+	dirname, filename := path.Split(strings.TrimSuffix(r.URL.Path, "/"))
+	dirname = strings.TrimSuffix(dirname, "/")
+	remoteIP := net.ParseIP(strings.Split(r.RemoteAddr, ":")[0])
+	var hostname string
+	var id string
+	anStatus := lookupAppNetworkStatusByAppIP(hdl.ctx, remoteIP)
+	if anStatus != nil {
+		hostname = anStatus.DisplayName
+		id = anStatus.UUIDandVersion.UUID.String()
+	} else {
+		log.Errorf("No AppNetworkStatus for %s",
```

Comment:
Can't we add the error string to the http response?

Reply:
It seems, from the point of view of the caller (cloud-init), the error is irrelevant.

Reply:
Sure, but if we need to debug some customer issue it might be useful to send a string? 
I don't know if the cloud-init scripts would ever display it, but even if not being able to invoke curl manually it would be helpful.